utils/build/build.mk

```diff
@@ -0,0 +1,71 @@
+# This makefile configures variables related to compilation, e.g. compiler flags.
+# Furthermore, it defines compilation recipes
+CFLAGS   := $(CFLAGS) -pthread -Wall
+LFLAGS   := $(LFLAGS) -lpthread -pthread
+CC       := g++
+INC_DIR  := $(INC_DIR) -I$(SRCS_DIR) -I$(LIBS_DIR)
+DEPFLAGS  = -MT $@ -MMD -MP -MF $(OBJS_DIR)/$*.d
```

Comment:
This is to let `make` know what needs to be recompiled. When you modify a header file, you really should recompile all object files that depend on the header file. This means not only .cpp files including the header file, but .cpp file including other headers that can eventually include the modified header file. Since the dependency tree is hard to track (for makefile), and is language (c++) specific, `DEPFLAGS` are passed to the compiler to generate those `.d` files. Those `.d` are formatted such, that they create dependency targets when included into `Makefile`. We include all of those here `Makefile:92: -include $(OBJS:.o=.d)`.

The `-M<...>` options are magic I do not entirely understand, but it works :)

To me, it seems confusing to put this explanation in the makefile. Can you suggest a way to document this? Updating the FAQ on Makefiles? It it even necessary to document?

Reply:
I added the [question](https://github.com/Hyp-ed/hyped-2020/wiki/Makefiles#what-is-the-line-depflags----mt---mmd--mp--mf-objs_dird-in-buildmk) to the wiki. I copied your answer above over there so feel free to modify it if you think anything could be improved.
I also suggested adding a small comment pointing to the wiki. But you might have a better formulation for that as well so feel free to change it.